### ISSUE TITLE
Code Quality Improvements

### ISSUE THREAD

**Issue by konstantinjdobler** [konstantinjdobler/gan-n1#4]
- ~move printing etc. code form training loop into dedicated functions~
- better loss printing / plot losses after training
  - I think loss is currently bugged for visualisation purposes, we should average the loss across all batches in an epoch 
- we now have a lot of arguments, move argparsing to different file
